### ISSUE TITLE
WS upload timeout blocks subsequent upload to CS.

### ISSUE THREAD

**Issue by bb30994** [FoldingAtHome/fah-client-pub#1167]
In the case below, the Donor's line-speed is limited and the upload packet is too large for the upload to be completed before the server timeout of 15 Minutes.  The CS has a longer timeout and the WU uploads successfully, but it is discarded with messages  
Server responded PLEASE_WAIT (464)
WARNING ... Failed to send results, will try again later.

When an upload to the WS times out, the subsequent upload to the CS must not be treated as an attempt to get credit for the same WU twice.
![capture](https://cloud.githubusercontent.com/assets/5483435/15224093/7d1f40ce-182c-11e6-8574-5e6c2e78873f.PNG)

**Comment by bb30994** [comment]
(When the first upload starts, acceptance of the same WU from the CS _should_ be blocked, but when the first upload times out, it needs to unblock the acceptance of the same WU from the CS... and that's probably what the message "PLEASE_WAIT + try later" means.)

**Comment by jcoffland** [comment]
@bb30994 There is no such blocking of WU uploads.

**Comment by bb30994** [comment]
The first upload was blocked by the 15 minute timeout.

The second upload succeeded but points could not be credited.

Those messages mean _SOMETHING_  What else might be the error that causes the rejection of the WU's credit?. 


Bruce
-
-
-


---------- Original Message ----------

**Comment by bb30994** [comment]
Please assign this post to whomever is responsible for the design of the points crediting system.

**Comment by jcoffland** [comment]
``PLEASE_WAIT`` is returned by the WS/CS when it's not ready to accept a WU.  ``Failed to send results...`` is printed by the client indicating that I was not able to upload due to the WS/CS returning ``PLEASE_WAIT``.  This is all normal behavior.  If a CS always returns ``PLEASE_WAIT`` then either it or the WS are misconfigured.

**Comment by jcoffland** [comment]
This is definitely not a client but it looks like a WS misconfiguration to me.  It should be reported to the WS manager.

**Comment by bb30994** [comment]
I only see this case when (A) both the WS and CS are functioning normally and (B) The upload of the WU to the WS was interrupted and (C) The upload to the CS succeeded.

When a WU has been placed in a WAIT state on the CS, what is supposed to happen to resolve that wait?   If the WAIT condition can be closed when the first upload is terminated, then the WU that was successfully uploaded to the CS should be eligible for acceptance, but that's not happening.

Previously there was a proposal to intentionally configure the CS to have a longer timeout than the WS so in cases like this where a slow client line causes the upload to be aborted, the WU could be successfully accepted by the CS.

In other words, the CS is currently unable to step in and provide an acceptable bypass for whatever caused the upload to the WS to be interrupted.  Why shouldn't that work?

**Comment by jcoffland** [comment]
@bb30994 It should work.  Despite the apparent correlation of the WS and CS not working at the same time I believe these are two separate issues caused by misconfiguration or mismanagement of the WS, CS or both.  This should be reported to the WS managers.

It's important to note that the WS does not tell the CS what to do with incoming WUs.  There is no such thing as the CS putting WUs on hold.
